paper/src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/basic/ArmourBound.java

```diff
@@ -72,6 +75,43 @@ public void onArmourEquip(PlayerArmorChangeEvent event){
 		}
 	}
 
+	@EventHandler
+	public void onItemCraft(PrepareItemCraftEvent event) {
+		CraftingInventory inventory = event.getInventory();
+		ItemStack[] matrix = inventory.getMatrix();
+		if (event.getRecipe() == null) {
+			return;
+		}
+		ItemStack boundItem = null;
+		PersistentDataContainer container = null;
+		ItemMeta meta = null;
+		for (ItemStack item : matrix) {
+			if (item == null) {
+				continue;
+			}
+			if (!whitelist.contains(item.getType().toString())) {
+				continue;
+			}
+			if (!item.hasItemMeta()) {
+				continue;
+			}
+			meta = item.getItemMeta();
+			container = meta.getPersistentDataContainer();
+			if (!container.has(this.key, PersistentDataType.STRING)) {
+				continue;
+			}
+			boundItem = item;
+			break;
+		}
+		if (boundItem == null || container == null) {
+			return;
+		}
```

Comment:
`container` will never be null since `meta.getPersistentDataContainer()` is annotated as `@NotNull` ([Reference](https://jd.papermc.io/paper/1.18/org/bukkit/persistence/PersistentDataHolder.html#getPersistentDataContainer()))